crates/ruff_linter/src/rules/pyupgrade/rules/deprecated_import.rs

```diff
@@ -342,11 +342,13 @@ const TYPING_EXTENSIONS_TO_TYPING_312: &[&str] = &[
     "SupportsFloat",
     "SupportsInt",
     "SupportsRound",
+    "TypeAliasType",
     "TypedDict",
```

Comment:
I'd also remove both this and `NamedTuple` from the list. We've backported new features to both recently that people might reasonably want to play around with, but won't be able to on py312 or lower:

- https://github.com/python/typing_extensions/commit/4f91502281d748671c7c1dfa26726111853f1342
- https://github.com/python/typing_extensions/commit/0b0166d649cebcb48e7e208ae5da36cfab5965fe